File: idea-plugin/src/main/java/ru/mipt/acsl/decode/parser/psi/DecodeTokenType.java
package ru.mipt.acsl.decode.parser.psi;

import ru.mipt.acsl.decode.parser.DecodeLanguage;
import com.intellij.psi.tree.IElementType;
import org.jetbrains.annotations.NotNull;

/**
 * @author Artem Shein
 */
public class DecodeTokenType extends IElementType
{
    public DecodeTokenType(@NotNull String debugName)
    {
        super(debugName, DecodeLanguage.INSTANCE);
    }

    @Override
    public String toString()
    {
        return "DecodeTokenType." + super.toString();
    }
}

File: parser/src/main/scala/ru/mipt/acsl/decode/parser/psi/DecodeTokenType.scala
package ru.mipt.acsl.decode.parser.psi

import ru.mipt.acsl.decode.parser.DecodeLanguage
import com.intellij.psi.tree.IElementType

/**
 * @author Artem Shein
 */
class DecodeTokenType(debugName: String) extends IElementType(debugName, DecodeLanguage.instance)
{
    override def toString: String = "DecodeTokenType." + super.toString
}
